x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/FillMaskConfig.java

```diff
@@ -214,8 +224,24 @@ public FillMaskConfig.Builder setResultsField(String resultsField) {
             return this;
         }
 
-        public FillMaskConfig build() {
+        public FillMaskConfig.Builder setMaskToken(String maskToken) {
+            this.maskToken = maskToken;
+            return this;
+        }
+
+        public FillMaskConfig build() throws IllegalArgumentException {
+            validateMaskTokenAgainstTokenizer(maskToken);
             return new FillMaskConfig(vocabularyConfig, tokenization, numTopClasses, resultsField);
         }
+
+        private void validateMaskTokenAgainstTokenizer(String maskToken) throws IllegalArgumentException {
+            String tokenizationMaskToken = tokenization == null ? "tokenization was unexpectedly null" : tokenization.getMaskToken();
```

Comment:
The logic here when a tokenization is not specified must match that in the constructor of the `FillMaskConfig` class:

```suggestion
            String tokenizationMaskToken = tokenization == null ? tokenization.createDefault().getMaskToken() : tokenization.getMaskToken();
```

But then, constructing the default tokenization for no reason is wasteful, this line should only be run if `maskToken != null`, which is very unlikely. So please restructure the `if` below into two so that this line can go in between them.